cli/config.go

```diff
@@ -577,9 +577,51 @@ func loadConfiguration(configPath string, ignoreLogging bool) (resolvedConfigPat
 		config.ProxyConfig = vc.ProxyConfig{}
 	}
 
+	if err := checkHypervisorConfig(config.HypervisorConfig); err != nil {
+		return "", config, err
+	}
+
 	return resolved, config, nil
 }
 
+// checkHypervisorConfig performs basic "sanity checks" on the hypervisor
+// config.
+func checkHypervisorConfig(config vc.HypervisorConfig) error {
+	memSizeMB := int64(config.DefaultMemSz)
+
+	if memSizeMB == 0 {
+		return errors.New("VM memory cannot be zero")
+	}
+
+	mb := int64(1024 * 1024)
+
+	for _, image := range []string{config.InitrdPath, config.ImagePath} {
+		if image == "" {
+			continue
+		}
+
+		imageSizeBytes, err := fileSize(image)
+		if err != nil {
+			return err
+		}
+
+		if imageSizeBytes == 0 {
+			return fmt.Errorf("image %q is empty", image)
+		}
+
+		if imageSizeBytes > mb {
+			imageSizeMB := imageSizeBytes / mb
+
+			if imageSizeMB >= memSizeMB {
```

Comment:
I didn't test more on this, can this work if `(imageSizeMB+1) == memSizeMB`? Should we leave some MB for applications?

for example: 

`if imageSizeMB >= memSizeMB + 64MB`

Reply:
I did think about that but thought I'd leave this PR as a very basic heuristic. I'm not sure how we can reliably determine "how much more" memory we need so kept it simple for this first PR.

Reply:
Any thoughts on this @grahamwhaley, @bergwolf, @sboeuf?

Reply:
I don't think we can guess how much memory a container app would use. In stead, we should try to catch OOM events from guest.

Reply:
`imageSize` is the size in bytes of the configured image type (`kata-containers.img` or `kata-containers-initrd.img`).

And yes, as already mentioned, this is a very simple check - we need something better going forward but this will catch some simple misconfiguration cases.

Reply:
So, yes, I think as a simple premise to catch the basic simple cases, this will work:
> If your VM RAM is < your kata.img size, you will always fail to load the VM

Reply:
As what I test and find..  ramdisk size might take a little trick here. Ramdisk size is half of vm memory and as we tested before. If   initrd not gizpped before take 100M, U at least need 200M vm memory. So if we just judge the imgSize, we will get an error of `Ramdisk doesn't exsit ` or other boot error.

@jodh-intel @grahamwhaley @bergwolf 
Tell me sth deeper if you guys have any ideas on this ~

Reply:
@jshachm Except for the space consumed by unpressed initramfs, you also need memory for the guest kernel, and all the stack/heap needed to run them. So I don't think you can simply guess a minimal guest memory size and expect all the binaries to run w/o issues.

@grahamwhaley 
> If your VM RAM is < your kata.img size, you will always fail to load the VM

This only applies to initramfs. For disk image, it is not fully loaded into the memory. Besides we plug it as a nvdimm device so if dax is on, it has very small memory footprint (for the file system metadata) in the guest.